### PR TITLE
Unified gRPC server initialization

### PR DESCRIPTION
**Pull request by jeremyje**

### CODE REVIEW

File: internal/serving/testing/minimatch.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+			GrpcServer: grpcServer,
+			Logger:     logger,
+			RedisPool:  pool,
+		},
+		mRedis: mredis,
+	}
+	for _, params := range paramsList {
+		for _, binding := range params.Bindings {
+			err := binding(mmServer.OpenMatchServer)
+			if err != nil {
```
Comment (Laremere) [inline internal/serving/testing/minimatch.go]: is there anything which needs to be stopped if an error occurs here?

File: internal/serving/testing/nextport.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+)
+
+const (
+	firstPort   = 10000
+	lastPort    = 60000
+	maxAttempts = 1000
+)
+
+var (
+	currentPort = int32(firstPort)
```
[the remark this answers is not in the document]
Comment (jeremyje) [reply internal/serving/testing/nextport.go]: I don't think it's worth having a long lived channel for this specially when it's just done on initialization.

File: internal/serving/testing/minimatch_test.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	if err != nil {
+		t.Errorf("could not create Mini Match context %s", err)
+	}
+	err = mm.Start()
+	if err != nil {
+		t.Errorf("could not start Mini Match %s", err)
+	}
+	defer func() {
+		mm.Stop()
+		/*
```
[the remark this answers is not in the document]
Comment (jeremyje) [reply internal/serving/testing/minimatch_test.go]: This is part of the test but it triggers an infinite loop.

File: internal/serving/testing/minimatch.go (hunk 1 above)
Comment (jeremyje) [reply internal/serving/testing/minimatch.go]: Added a fix for this and TODO for a cleaner solution in the next PR.

**Pull request merged**